examples/utils/bundle-size-tests/webpack.config.js

```diff
@@ -30,6 +49,17 @@ module.exports = {
 				use: "ts-loader",
 				exclude: /node_modules/,
 			},
+			{
+				test: /\.js$/,
```

Comment:
Disclaimer: I'm not the most well versed in webpack.

We can't be more specific and target `packageVersion.js` here? Is this already working on post-bundling files?

Reply:
yeah, that should be doable provided this is the only place the version string shows up (I think that's true). Loaders are chained and invoked [bottom-to-top](https://webpack.js.org/concepts/loaders/#configuration) as middleware during bundling.

This webpack config points at a bunch of entrypoint files which transitively reference code that's already been compiled in dependent packages. So this config should work fine. But if a package that had a `packageVersion.ts` was additionally an application (we have an example package like this--webflow I think), we'd want to put this rule before `ts-loader` in the list, since usually webpack configs will target source files for the package they live in.

Reply:
> I'd lean toward the latter being more of a danger, as once we get off the 2.0.0-internal versioning scheme, there are some other places (usually with protocol/persistent versioning concerns) that people might happen to choose semver-like constructs that have overlap with something we're replacing. It's still pretty minor though, since when we're replacing a version, the string we're searching for has something like a build hash on it (and not just '1.0.0' -> '1.0.0-internal' or something likely to collide).

I agree with your reasoning. Another mitigating factor here is that this webpack output is not used other than to test bundle sizes. So if we do manage to get the versions wrong it should only lead to confusing +/-1-3 byte differences in the bundle size tests. :smile:

> So this config should work fine. But if a package that had a `packageVersion.ts` was additionally an application (we have an example package like this--webflow I think), we'd want to put this rule before `ts-loader` in the list, since usually webpack configs will target source files for the package they live in.

It sounds like it's best to both narrow the `test` condition to packageVersion.js, and also move it to be before `ts-loader` (and thus run after it). I made those changes in the latest iteration. Let me know if I misunderstood.

> Loaders are chained and invoked [bottom-to-top](https://webpack.js.org/concepts/loaders/#configuration) as middleware during bundling.

The docs seemed ambiguous to me at first. The example they give has a single rule that invokes multiple loaders, while this config declares multiple rules with a loader defined. But then I read this: https://webpack.js.org/configuration/module/#ruleloader

> `Rule.loader` is a shortcut to `Rule.use: [ { loader } ]`.

So I guess the example is equivalent to ours.